core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -90,11 +90,26 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
   def flatMapF[AA >: A, D](f: B => F[Either[AA, D]])(implicit F: Monad[F]): EitherT[F, AA, D] =
     flatMap(f andThen EitherT.apply)
 
+  def subflatMap[AA >: A, D](f: B => Either[AA, D])(implicit F: Functor[F]): EitherT[F, AA, D] =
+    transform(_.flatMap(f))
+
+  def semiflatMap[D](f: B => F[D])(implicit F: Monad[F]): EitherT[F, A, D] =
+    flatMap(b => EitherT.right(f(b)))
+
   def transform[C, D](f: Either[A, B] => Either[C, D])(implicit F: Functor[F]): EitherT[F, C, D] =
     EitherT(F.map(value)(f))
 
-  def subflatMap[AA >: A, D](f: B => Either[AA, D])(implicit F: Functor[F]): EitherT[F, AA, D] =
-    transform(_.flatMap(f))
+  def flatTap[AA >: A, D](f: B => EitherT[F, AA, D])(implicit F: Monad[F]): EitherT[F, AA, B] =
```

Comment:
I think this one is already provided by `Monad` :)

Reply:
Please look at the open questions in issue #2440. `flatTap` is provided by Monad but for EitherT it doesn't support widening of the second type parameter.

Reply:
This version allows widening of A. `Monad`'s one doesn't.

Reply:
Will this hide the `flatTap` syntax provided by `Monad`? The type parameters are different. It might break source compatibility, I haven't tested it though so I could be wrong.

Reply:
Thanks, will check this

Reply:
Tried to compose some examples. Can't imagine what can break source compatibility here. Any ideas on what to check? Do you mean type inference might break?

Hiding may cause `unused import` but it's irrelevant to widening.

Reply:
The `flatTap` syntax provided by `Monad` take one type parameters, while this one takes two. Both could be inferred, in which case they are source compatible. But there might be use sites where the type parameter of `flatTap` is provided, with this new version, it would no longer compile, right?

Reply:
Yes, you're right. Does source breaking require a major version upgrade?

Reply:
sorry I dropped the ball on this. It's a rather minor source breaking change, I wouldn't say we have to bump the major version. I am okay to have this merge as is, but I'd like to hear from other maintainers. @LukaJCB?

Reply:
@LukaJCB @kailuowang Ok, source compatibility is important. But instead of removing I would rather keep `flatTap` in data types with the same signature as in `FlatMap.Ops` (`def flatTap[B](f: A => F[B]): F[A]`):
- pros: it can use the optimized `as`, while the syntax extension cannot, since we didn't override `as` in `Functor` instances due to bincompat issues
- cons: its signature will be different from other similar functions in the `flat` family for some data types (no widening of the left type param). and this probably needs a TODO comment